src/panels/lovelace/editor/hui-dialog-edit-card.ts

```diff
@@ -53,58 +64,115 @@ export class HuiDialogEditCard extends LitElement {
     return this.shadowRoot!.querySelector("hui-yaml-card-preview")!;
   }
 
-  protected render() {
+  protected render(): TemplateResult {
     return html`
       <style>
         paper-dialog {
           width: 650px;
         }
+        .element-editor {
+          margin-bottom: 16px;
+        }
       </style>
       <paper-dialog with-backdrop>
         <h2>Card Configuration</h2>
         <paper-dialog-scrollable>
-          <hui-yaml-editor
-            .yaml="${this._cardConfig}"
-            @yaml-changed="${this._handleYamlChanged}"
-          ></hui-yaml-editor>
+          ${
+            this._editorToggle && this._elementConfig !== null
+              ? html`<div class="element-editor">${when(
+                  this._elementConfig,
+                  () => this._elementConfig,
+                  () => html`Loading...`
+                )}</div>`
+              : html`
+              <hui-yaml-editor
+                .yaml="${this._cardConfig}"
+                @yaml-changed="${this._handleYamlChanged}"
+              ></hui-yaml-editor>`
+          }
           <hui-yaml-card-preview
             .hass="${this.hass}"
             .yaml="${this._cardConfig}"
           ></hui-yaml-card-preview>
         </paper-dialog-scrollable>
         <div class="paper-dialog-buttons">
-          <paper-button @click="${this._closeDialog}">Cancel</paper-button>
-          <paper-button @click="${this._updateConfig}">Save</paper-button>
+          <paper-button
+            @click="${this._toggleEditor}"
+          >Toggle Editor</paper-button>
+          <paper-button
+            @click="${this._closeDialog}"
+          >Cancel</paper-button>
+          <paper-button
+            @click="${this._updateConfig}"'
+          >Save</paper-button>
         </div>
       </paper-dialog>
     `;
   }
 
-  private _handleYamlChanged(ev) {
-    this._previewEl.yaml = ev.detail.yaml;
+  protected updated(): void {
+    // This will center the dialog with the updated config
+    fireEvent(this._dialog, "iron-resize");
+  }
+
+  private _handleYamlChanged(ev: MouseEvent): void {
+    this._handleConfigChanged("yaml", (ev.detail as any).yaml);
   }
 
-  private _closeDialog() {
+  private _handleConfigChanged(
+    format: string,
+    value: LovelaceConfig | string
+  ): void {
+    if (!this._previewEl) {
+      return;
+    }
+
+    if (format === "js") {
+      this._newConfigYaml = yaml.safeDump(value);
```

Comment:
Two things are weird here:
 - why is this only set when it's JS and not when YAML ? The name `_handleConfigChanged` seems like the method that should always set it, no other methods.
 - We should convert every change to YAML. Do less and be lazy: we only need it as YAML when the user presses thet save button. We should store the exact same format as that we pass the preview element.

Reply:
This removes the need for configValue. I can just always update the YAML and send the Yaml to the backend and the review. So now it knows its always YAML